modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/utils.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+export default function parseAssignments(node) {
+	const assignments = {};
+	const users = [];
+
+	node.assignments.forEach((item) => {
+		for (const [key, value] of Object.entries(item)) {
+			if (key === 'user') {
+				assignments.assignmentType = ['user'];
+			}
+			if (key === 'resource-action') {
```

Comment:
@cgmonte use else if in the following verifications to avoid unnecessary steps in the code execution.